### PR TITLE
reference: update constraints.md

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### What is changed, added or deleted? (Required)

1. Reorder the list of constraint, not null, unique, primary, foreign key as other databases.
1. Add unique info in primary.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: constraints.md [hunk 1]
```diff
@@ -6,67 +6,7 @@ aliases: ['/docs-cn/dev/reference/sql/constraints/']
...
-* TiDB 支持外键是为了在将其他数据库迁移到 TiDB 时，不会因为此语法报错。但是，TiDB 不会在 DML 语句中对外键进行约束。例如，即使 `users` 表中不存在 `id=123` 的记录，下列事务也能提交成功：
-
-    ```
-    START TRANSACTION;
-    INSERT INTO orders (user_id, doc) VALUES (123, NULL);
-    COMMIT;
-    ```
-
-* TiDB 在执行 `SHOW CREATE TABLE` 语句的结果中不显示外键信息。
+TiDB 支持的约束与 MySQL 的基本相同。
```
Comment (kissmydb) [inline constraints.md:9]:
```suggestion
TiDB 支持的约束与 MySQL 的基本相同。不同约束的详细描述如下：
```

File: constraints.md [hunk 2]
```diff
@@ -118,61 +58,67 @@ Query OK, 1 row affected (0.03 sec)
 
 * 第三条 `INSERT` 语句成功，因为 `last_login` 列没有被明确地指定为 `NOT NULL`。默认允许 `NULL` 值。
 
-## 主键约束
+## 唯一约束
 
-TiDB 支持的主键约束规则与 MySQL 支持的相似。例如：
+在 TiDB 中，默认会对唯一约束进行[惰性检查](/transaction-overview.md#事务的惰性检查)。通过在事务提交时再进行批量检查，TiDB 能够减少网络开销、提升性能。例如：
```
Comment (kissmydb) [inline constraints.md:63]:
```suggestion
在 TiDB 中，默认会对唯一约束进行[惰性检查](/transaction-overview.md#事务的惰性检查)。惰性检查会在事务提交时批量进行唯一检查，能够减少 TiDB 的网络开销、提升性能。例如：
```

File: constraints.md [hunk 3]
```diff
@@ -118,61 +58,67 @@ Query OK, 1 row affected (0.03 sec)
...
+COMMIT;
+```
 
-## 唯一约束
+```
+ERROR 1062 (23000): Duplicate entry 'bill' for key 'username'
+```
 
-在 TiDB 中，默认会对唯一约束进行惰性检查。通过直到事务提交时才进行批量检查，TiDB 能够减少网络通信开销。例如：
+* 第一条 `INSERT` 语句不会导致重复键错误，这同 MySQL 的规则一致。该检查将推迟到事务提交时才会进行。
```
Comment (yikeke) [inline constraints.md:119]:
```suggestion
第一条 `INSERT` 语句不会导致重复键错误，这同 MySQL 的规则一致。该检查将推迟到事务提交时才会进行。
```
列表项只有一个的话，就不建议用列表格式了

File: constraints.md [hunk 4]
```diff
@@ -118,61 +58,67 @@ Query OK, 1 row affected (0.03 sec)
...
 
-{{< copyable "sql" >}}
+您可通过设置 `tidb_constraint_check_in_place` 为 `1` 停用此行为。如果停用此行为，则会在执行语句时就对唯一约束进行检查。例如：
```
Comment (yikeke) [inline constraints.md:121]:
```suggestion
你可通过设置 `tidb_constraint_check_in_place` 为 `1` 停用此行为。如果停用此行为，则会在执行语句时就对唯一约束进行检查。例如：
```
官网用户文档中统一用「你」，不用「您」

File: constraints.md [hunk 5]
```diff
@@ -197,78 +143,128 @@ Query OK, 0 rows affected (0.00 sec)
...
 ```
 
 ```
-Query OK, 2 rows affected (0.00 sec)
-Records: 2  Duplicates: 0  Warnings: 0
+ERROR 1062 (23000): Duplicate entry 'bill' for key 'username'
+..
 ```
 
+* 第一条 `INSERT` 语句导致了重复键错误。这会造成额外的网络通信开销，并可能降低插入操作的吞吐量。
```
Comment (yikeke) [inline constraints.md:164]:
```suggestion
第一条 `INSERT` 语句导致了重复键错误。这会造成额外的网络通信开销，并可能降低插入操作的吞吐量。
```

File: constraints.md [hunk 6]
```diff
@@ -197,78 +143,128 @@ Query OK, 0 rows affected (0.00 sec)
...
+
+## 主键约束
+
+首先，与 MySQL 相同，主键约束包含了唯一约束，即创建了主键约束相当于拥有了唯一约束，此外，TiDB 其他的主键约束规则与 MySQL 相似。例如：
```
Comment (yikeke) [inline constraints.md:168]:
```suggestion
与 MySQL 行为一样，主键约束包含了唯一约束，即创建了主键约束相当于拥有了唯一约束。此外，TiDB 其他的主键约束规则也与 MySQL 相似。例如：
```

File: constraints.md [hunk 7]
```diff
@@ -197,78 +143,128 @@ Query OK, 0 rows affected (0.00 sec)
...
-START TRANSACTION;
+CREATE TABLE t4 (a INT NOT NULL, b INT NOT NULL, PRIMARY KEY (a,b));
 ```
 
 ```
-Query OK, 0 rows affected (0.00 sec)
+Query OK, 0 rows affected (0.10 sec)
+```
+
+* 表 `t2` 创建失败，因为定义为主键的列 `a` 不能允许 `NULL` 值。
```
Comment (yikeke) [inline constraints.md:210]:
```suggestion
分析：

* 表 `t2` 创建失败，因为定义为主键的列 `a` 不能允许 `NULL` 值。
```

**Comment by yikeke** [comment]
Please tick the checkboxes under `Which TiDB version(s) do your changes apply to?` in the PR description box. @jackysp

File: constraints.md (hunk 1 above)
Comment (jackysp) [reply constraints.md:9]: We don't have the part for the different constraints.

File: constraints.md [hunk 8]
```diff
@@ -197,78 +143,128 @@ Query OK, 0 rows affected (0.00 sec)
...
+* 表 `t3` 创建失败，因为一张表只能有一个主键。
+* 表 `t4` 创建成功，因为虽然只能有一个主键，但 TiDB 支持定义一个多列组合作为复合主键。
+
+除上述规则外，TiDB 还强加了另一个限制，即一旦一张表创建成功，其主键就不能再改变。
+
+## 外键约束
+
+TiDB 支持创建外键约束。例如：
```
Comment (kissmydb) [inline constraints.md:220]:
```suggestion
TiDB 在语法上支持创建外键约束。例如：
```

Comment (jackysp) [reply constraints.md:220]: I think it does not only "仅在语法上支持". The created foreign key could be written in the table metadata, and it even could be dropped. I think "部分支持“ is better.

Comment (kissmydb) [reply constraints.md:220]: ok

**Comment by coocood** [comment]
LGTM

File: constraints.md [hunk 9]
```diff
@@ -197,78 +143,134 @@ Query OK, 0 rows affected (0.00 sec)
...
+Query OK, 0 rows affected (0.10 sec)
+```
+
+分析：
+
+* 表 `t2` 创建失败，因为定义为主键的列 `a` 不能允许 `NULL` 值。
+* 表 `t3` 创建失败，因为一张表只能有一个主键。
+* 表 `t4` 创建成功，因为虽然只能有一个主键，但 TiDB 支持定义一个多列组合作为复合主键。
+
+除上述规则外，TiDB 还强加了另一个限制，即一旦一张表创建成功，其主键就不能再改变。
```
Comment (AilinKid) [inline constraints.md:216]: alter primary key 要稍提一下吗

Comment (yikeke) [reply constraints.md:216]: PTAL @jackysp

Comment (jackysp) [reply constraints.md:216]: 可以

Comment (jackysp) [reply constraints.md]: PTAL @yikeke @AilinKid

**Pull request merged**